File: solns/clojure/5.clj
(defn gcd
  ([a b] (if (zero? b) a (gcd b (rem a b))))
  ([a b & more] (reduce gcd (conj more (gcd a b)))))

(defn lcm
  ([a b] (* a (/ b (gcd a b))))
  ([a b & more] (reduce lcm (conj more (lcm a b)))))

(def result (reduce lcm (range 1 20)))

(println result)
(assert (= result 232792560))

File: solns/ruby/5.rb
def gcd(a, b)
  b.zero? ? a : gcd(b, a % b)
end

def lcm(a, b)
  a * (b / gcd(a, b))
end

result = (1..20).inject(20) { |a, b| lcm(a, b) }

puts result
raise Error unless result == 232_792_560
